packages/performance-tests/test/TestContext.ts

```diff
@@ -0,0 +1,84 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { BriefcaseDb, BriefcaseManager, IModelHost, RequestNewBriefcaseArg } from "@itwin/core-backend";
+import { Logger } from "@itwin/core-bentley";
+import { BriefcaseIdValue } from "@itwin/core-common";
+import { AccessTokenAdapter, BackendIModelsAccess } from "@itwin/imodels-access-backend";
+import assert from "assert";
+
+const loggerCategory = "TestContext";
+
+export interface TestIModel {
+  name: string;
+  iModelId: string;
+  iTwinId: string;
+  tShirtSize: string;
+  load: () => Promise<BriefcaseDb>;
+}
+
+export const testITwinIds = [
+  "892aa2c9-5be8-4865-9f37-7d4c7e75ebbf",
+  // "523a1365-2c85-4383-9e4c-f9ec25d0e107",
+  // "bef34215-1046-4bf1-b3be-a30ae52fefb6",
+];
+
+type TShirtSize = "s" | "m" | "l" | "xl" | "unknown";
+
+function getTShirtSizeFromName(name: string): TShirtSize {
+  return /^(?<size>s|m|l|xl)\s*-/i.exec(name)?.groups?.size?.toLowerCase() as TShirtSize ?? "unknown";
+}
+
+export async function *getTestIModels() {
+  assert(IModelHost.authorizationClient !== undefined);
+  const hubClient = (IModelHost.hubAccess as BackendIModelsAccess)["_iModelsClient"]
+
+  for (const iTwinId of testITwinIds) {
+    const iModels = hubClient.iModels.getMinimalList({
+      authorization: AccessTokenAdapter.toAuthorizationCallback(
+        await IModelHost.authorizationClient.getAccessToken()
+      ),
+      urlParams: { projectId: iTwinId },
+    });
+
+    for await (const iModel of iModels) {
+      const iModelId = iModel.id;
+      yield {
+        name: iModel.displayName,
+        iModelId,
+        iTwinId,
+        tShirtSize: getTShirtSizeFromName(iModel.displayName),
+        load: async () => downloadAndOpenBriefcase({ iModelId, iTwinId }),
+      };
+    }
+  }
+}
+
+export async function downloadAndOpenBriefcase(briefcaseArg: Omit<RequestNewBriefcaseArg, "accessToken">): Promise<BriefcaseDb> {
+  const PROGRESS_FREQ_MS = 2000;
+  let nextProgressUpdate = Date.now() + PROGRESS_FREQ_MS;
+
+  assert(IModelHost.authorizationClient !== undefined, "auth client undefined");
+  const briefcaseProps =
+    BriefcaseManager.getCachedBriefcases(briefcaseArg.iModelId)[0] ??
```

Comment:
I just ran into an issue with this code in the place from which I copied it, not that it matters that much in our current tests, but the cached iModel may not match the `briefcaseArg`, specifically which changeset it's on. You should probably try to find a cached briefcase with the same changeset id, and if not, always download.

Reply:
I think this fixes that potential issue, but don't know if this solution is over complicated.

Reply:
I am resolving this in favor of new comments on the changes